### PR TITLE
main: improve error handling when loading target/*.json

### PR DESCRIPTION
**Pull request by sago35**
`tinygo targets` was failing and has been corrected.
Improv error messages to determine the error location.
Add run `tinygo targets` to CI.

### CODE REVIEW

**Comment by sago35** [comment]
This PR makes error messages easier to understand.

before:
```
panic: duplicate value 'esp32c3' in field : BuildTags

goroutine 1 [running]:
github.com/tinygo-org/tinygo/compileopts.(*TargetSpec).overrideProperties(0xc000353600, 0xc000353340)
        C:/tinygo/tinygo/compileopts/target.go:98 +0x848
github.com/tinygo-org/tinygo/compileopts.(*TargetSpec).resolveInherits(0xc000353340)
        C:/tinygo/tinygo/compileopts/target.go:157 +0x4d
github.com/tinygo-org/tinygo/compileopts.LoadTarget(0xc0000c9df0)
        C:/tinygo/tinygo/compileopts/target.go:223 +0x2b3
main.main()
        C:/tinygo/tinygo/main.go:1607 +0x3b70
```

after:
```
could not list target: C:\tinygo\tinygo\targets\esp32c3-12f.json : duplicate value 'esp32c3' in field : BuildTags
```

File: compileopts/target.go [hunk 1]
```diff
@@ -95,14 +95,15 @@ func (spec *TargetSpec) overrideProperties(child *TargetSpec) {
...
 					w := dst.Index(j).String()
 					if v == w {
-						panic("duplicate value '" + v + "' in field : " + field.Name)
+						return fmt.Errorf("duplicate value '" + v + "' in field : " + field.Name)
 					}
 				}
 			}
 		default:
-			panic("unknown field type : " + kind.String())
+			return fmt.Errorf("unknown field type : " + kind.String())
```
Comment (aykevl) [inline compileopts/target.go]:
```suggestion
			return fmt.Errorf("unknown field type: %s", kind)
```

File: compileopts/target.go [hunk 2]
```diff
@@ -95,14 +95,15 @@ func (spec *TargetSpec) overrideProperties(child *TargetSpec) {
 				for j := i + 1; j < dst.Len(); j++ {
 					w := dst.Index(j).String()
 					if v == w {
-						panic("duplicate value '" + v + "' in field : " + field.Name)
+						return fmt.Errorf("duplicate value '" + v + "' in field : " + field.Name)
```
Comment (aykevl) [inline compileopts/target.go]:
```suggestion
						return fmt.Errorf("duplicate value '%s' in field %s", v, field.Name)
```

**Review by aykevl — COMMENTED** [review]
Good idea, thanks for the patch!

**Comment by sago35** [comment]
Thanks for the review.

**Comment by sago35** [comment]
Targets/*.json is to be changed in #3128 
Only error handling is changed in this PR.

**Comment by sago35** [comment]
After #3128 is merged, the CI for this PR will succeed.

**Comment by deadprogram** [comment]
@sago35 can you please rebase against `dev` now that #3128 has been merged? Thanks!

**Comment by sago35** [comment]
@deadprogram  rebased!

**Comment by deadprogram** [comment]
Thank you for this @sago35 the error message is really nice looking now! Also thanks @aykevl for helping get it done.

Now merging.

**Pull request merged**